service/assessment/assessment.go

```
@@ -98,60 +101,105 @@ func (s Service) AssessEvidences(stream assessment.Assessment_AssessEvidencesSer
 	}
 }
 
-func (s Service) handleEvidence(evidence *evidence.Evidence) (result *evidence.Evidence, err error) {
-	log.Infof("Received evidence for resource %s", evidence.ResourceId)
-	log.Debugf("Evidence: %+v", evidence)
+func (s Service) handleEvidence(evidence *evidence.Evidence) error {
+	if evidence.Resource == nil {
+		return errors.New("evidence must contain a valid resource")
+	}
 
-	var file string
+	value := evidence.Resource.GetStructValue()
+	if value == nil {
+		return errors.New("resource in evidence is not struct value")
+	}
 
-	listId++
-	evidence.Id = fmt.Sprintf("%d", listId)
+	m := evidence.Resource.GetStructValue().AsMap()
+	if m == nil {
+		return errors.New("resource in evidence is not a map")
+	}
 
-	if len(evidence.ApplicableMetrics) == 0 {
-		log.Warnf("Could not find a valid metric for evidence of resource %s", evidence.ResourceId)
+	field, ok := m["id"]
+	if !ok {
+		return errors.New("resource in evidence is missing the id field")
 	}
 
-	// TODO(oxisto): actually look up metric via orchestrator
-	for _, metric := range evidence.ApplicableMetrics {
-		var baseDir string = "."
+	resourceId, ok := field.(string)
+	if !ok {
+		return errors.New("resource id in evidence is not a string")
+	}
 
-		// check, if we are in the root of Clouditor
-		if _, err := os.Stat("policies"); os.IsNotExist(err) {
-			// in tests, we are relative to our current package
-			baseDir = "../../"
-		}
+	toolId := evidence.ToolId
+	// If no toolId was provided, indicate it in the info message
+	if toolId == "" {
+		toolId = "<Tool id missing>"
```

we should probably return an error then